docs/tutorial_ptk.rst

```diff
@@ -0,0 +1,139 @@
+.. _tutorial_ptk:
+
+***********************************************
+Tutorial: ``prompt_toolkit`` custom keybindings
+***********************************************
+
+Are you really jonesing for some special keybindings? We can help you out with
+that. The first time is free and so is every other time!
+
+.. warning:: This tutorial will let you hook directly into the
+             ``prompt_toolkit`` keybinding manager. It will not stop you from
+             rendering your prompt completely unusable, so tread lightly.
+
+
+Overview
+========
+
+The ``prompt_toolkit`` shell has a registry for handling custom keybindings. You
+may not like the default keybindings in xonsh, or you may want to add a new key
+binding.
+
+We'll walk you though how to do this using ``prompt_toolkit`` tools to define
+keybindings and warn you about potential pitfalls.
+
+All of the code below can be entered into your `xonshrc <xonshrc.html>`_
+
+Control characters
+==================
+
+We can't and won't stop you from doing what you want, but in the interest of a
+functioning shell, you probably shouldn't mess with the following keystrokes.
+Some of them are `ASCII control characters
+<https://en.wikipedia.org/wiki/Control_character#In_ASCII>`_ and _really_
```

Comment:
Githubs rst parser is woefully insufficient and probably not something we should really worry about here.